src/Zio/FileSystems/MemoryFileSystem.cs

```diff
@@ -21,14 +21,16 @@ public class MemoryFileSystem : FileSystem
 
         private readonly DirectoryNode _rootDirectory;
         private readonly FileSystemNodeReadWriteLock _globalLock;
+        private readonly FileSystemEventDispatcher<Watcher> _dispatcher;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="MemoryFileSystem"/> class.
         /// </summary>
         public MemoryFileSystem()
         {
-            _rootDirectory = new DirectoryNode();
+            _rootDirectory = new DirectoryNode(this);
             _globalLock = new FileSystemNodeReadWriteLock();
+            _dispatcher = new FileSystemEventDispatcher<Watcher>();
```

Comment:
Could we delay the creation of a _dispatcher only if we are actually using the watch API? I would prefer if we could avoid the cost of having a thread created...etc.